addons/payment_adyen/models/payment.py

```diff
@@ -27,6 +168,18 @@ class AcquirerAdyen(models.Model):
     adyen_skin_code = fields.Char('Skin Code', required_if_provider='adyen', groups='base.group_user')
     adyen_skin_hmac_key = fields.Char('Skin HMAC Key', required_if_provider='adyen', groups='base.group_user')
 
+    @api.model
+    def _adyen_convert_amount(self, amount, currency):
+        """
+        Adyen requires the amount to be multiplied by 10^k,
+        where k depends on the currency code.
+        """
+        if currency.name not in CURRENCY_CODE_MAPS:
+            raise UserError(_("This currency is currently unsupported by Adyen."))
+        k = CURRENCY_CODE_MAPS[currency.name]
```

Comment:
`k = CURRENCY_CODE_MAPS.get(currency.name, 2)` and you save a hundred lines.

Reply:
I prefer to have the explicit map, since any code editor can just fold the lines.
And that would avoid passing an arbitrary value if it is not in the code map.

For your other remark, I've tested what happens by forcing the exception, and you get a red frame displaying the error message.

But in both cases I can do the change if you want.

Reply:
I think it is very unlikely that a user would subscribe to Adyen if the currency he is using is not supported by the service. So I would remove the check.

I also think we should keep the diff minimal. With the `get()`, we have a good chance that a new currency supported by Adyen would work in Odoo too.